huaweicloud/data_source_huaweicloud_compute_instance.go

```diff
@@ -165,113 +168,63 @@ func DataSourceComputeInstance() *schema.Resource {
 	}
 }
 
-func dataSourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error {
-	config := meta.(*config.Config)
-	ecsClient, err := config.ComputeV1Client(GetRegion(d, config))
-	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud ECS client: %s", err)
-	}
-
-	epsID := GetEnterpriseProjectID(d, config)
-	listOpts := &cloudservers.ListOpts{
+func buildEcsListOpts(d *schema.ResourceData, conf *config.Config) *cloudservers.ListOpts {
+	result := cloudservers.ListOpts{
```

Comment:
suggest to use `listOpts`

Reply:
use `buildListOptsWithoutStatus` to replace it.